sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -234,7 +235,11 @@ impl InputSelection {
                 .unwrap()
                 .0;
 
-            self.addresses.contains(&required_address)
+            if let Address::Restricted(restricted_address) = required_address {
+                self.addresses.contains(restricted_address.address())
+            } else {
+                self.addresses.contains(&required_address)
+            }
```

Comment:
that's weird tbh , but I guess there's no other way?

Reply:
What is weird exactly?

Reply:
I would like to prevent doing the pattern matching on the user side of that enum, but instead let  the `Address` type have a method that returns the address to compare. Is that possible?